src/main/java/org/janelia/saalfeldlab/n5/GsonKeyValueN5Reader.java

```diff
@@ -98,7 +100,8 @@ default DataBlock<?> readBlock(
 		final String path = absoluteDataBlockPath(N5URI.normalizeGroupPath(pathName), gridPosition);
 
 		try (final LockedChannel lockedChannel = getKeyValueAccess().lockForReading(path)) {
-			return DefaultBlockReader.readBlock(lockedChannel.newInputStream(), datasetAttributes, gridPosition);
+			final ArrayCodec codec = datasetAttributes.getArrayCodec();
+			return codec.decode(ReadData.from(lockedChannel.newInputStream()), gridPosition);
```

Comment:
this should use the KVA#createReadData method; probably this PR should merge after #156